### PR TITLE
align with abstract

### PR DESCRIPTION
**Pull request by delcroip**

### CODE REVIEW

File: calcrule_validations/calculation_rule.py [hunk 1]
```diff
@@ -49,40 +49,48 @@ def ready(cls):
...
         field_value = kwargs.pop('field_value')
-        if cls.active_for_object(validation_class, calculation_uuid):
-            return cls.calculate(validation_class, field_name, field_value, **kwargs)
+        if cls.active_for_object(instance, calculation_uuid):
+            return cls.calculate(instance, field_name, field_value, **kwargs)
 
     @classmethod
-    def active_for_object(cls, validation_class, calculation_uuid):
-        return cls.check_calculation(validation_class, calculation_uuid)
+    def active_for_object(cls, instance, calculation_uuid=None):
```
Comment (weilu) [inline calcrule_validations/calculation_rule.py:76]: This is still not in line with what AbsStrategy requires: https://github.com/openimis/openimis-be-core_py/blob/develop/core/abs_calculation_rule.py#L102

File: calcrule_validations/calculation_rule.py [hunk 2]
```diff
@@ -49,40 +49,48 @@ def ready(cls):
...
+        return cls.check_calculation(instance, calculation_uuid)
 
     @classmethod
     def get_linked_class(cls, sender, class_name, **kwargs):
         return ["Calculation"]
 
     @classmethod
-    def check_calculation(cls, validation_class, calculation_uuid, **kwargs):
-        return ValidationStrategyStorage.choose_strategy(validation_class).check_calculation(cls, calculation_uuid)
+    def check_calculation(cls, instance, calculation_uuid=None, **kwargs):
```
Comment (weilu) [inline calcrule_validations/calculation_rule.py:86]: Still not meeting what AbsStrategy requires: https://github.com/openimis/openimis-be-core_py/blob/develop/core/abs_calculation_rule.py#L97

File: calcrule_validations/calculation_rule.py [hunk 3]
```diff
@@ -49,40 +49,48 @@ def ready(cls):
...
+        if not calculation_uuid:
+            return False
+        return ValidationStrategyStorage.choose_strategy(instance).check_calculation(cls, calculation_uuid)
 
     @classmethod
-    def calculate(cls, validation_class, field_name, field_value, **kwargs):
-        return ValidationStrategyStorage.choose_strategy(validation_class)\
+    def calculate(cls, instance, field_name=None, field_value=None, **kwargs):
```
Comment (weilu) [inline calcrule_validations/calculation_rule.py:92]: Is this change necessary? `AbsStrategy.calculate` takes `cls, instance, *args, **kwargs` which is already satisfied with the existing code: https://github.com/openimis/openimis-be-core_py/blob/develop/core/abs_calculation_rule.py#L107

**Review by weilu — CHANGES_REQUESTED** [review]
@delcroip can you add a test to demonstrate the issue caused by the existing implementation and that the changes proposed fixes the issue? It would also guard regression for future occurrences. Thanks!

File: calcrule_validations/calculation_rule.py (hunk 3 above)
Comment (delcroip) [reply calcrule_validations/calculation_rule.py:92]: yes but we should drop the *args in the ABS because it will still fail without that change even if was aligned

File: calcrule_validations/calculation_rule.py (hunk 2 above)
Comment (delcroip) [reply calcrule_validations/calculation_rule.py:86]: indeed we should have def check_calculation(cls, instance, *kwars) but if this calc is call with the abs def it will work 
let's say that I focused to make it work because of the urgency

File: calcrule_validations/calculation_rule.py (hunk 1 above)
Comment (delcroip) [reply calcrule_validations/calculation_rule.py:76]: indeed, it will not happen until someone will use this calculation elsewhere. As I explained to Damian we should drop the approach using the abstract and focus on Validation only (this is an helper to access BaseValidation) but this has impact on social protection so I did not want to do this kind on change.

**Comment by delcroip** [comment]
No I did not add test because they are already present :) BatchRun and Contract are passing with this PR

**Comment by delcroip** [comment]
I just realize that the CI is not started on this repo

**Review by sniedzielski — APPROVED** [review]
I agree with some points with @weilu, but due to urgency - let's merge. I tested locally and it works while uploading individuals/beneficiary.
I can confirm that tests exist in other calcrule related to 'batch run' for example as @delcroip said. 

@delcroip You can merge, please decide.

**Comment by weilu** [comment]
Ok, sounds good. Please proceed with merge and deployment.

**Review by weilu — APPROVED** [review]
